grid-client/subi/substrate_manager.go

```diff
@@ -79,6 +86,79 @@ type SubstrateImpl struct {
 	m sync.Mutex
 }
 
+var _ SubstrateExt = (*SubstrateImpl)(nil)
+
+// NewIdentityFromSr25519Phrase returns the identity from mnemonic
+func (s *SubstrateImpl) NewIdentityFromSr25519Phrase(mnemonic string) (substrate.Identity, error) {
+	return substrate.NewIdentityFromSr25519Phrase(mnemonic)
+}
+
+// AcceptTermsAndConditions accepts terms and conditions
+func (s *SubstrateImpl) AcceptTermsAndConditions(identity substrate.Identity, docLink string, docHash string) error {
+	s.m.Lock()
+	defer s.m.Unlock()
+	return s.Substrate.AcceptTermsAndConditions(identity, docLink, docHash)
+}
+
+// CreateTwin creates a twin and returns its twin ID
+func (s *SubstrateImpl) CreateTwin(identity substrate.Identity, relay string, pk []byte) (uint32, error) {
+	s.m.Lock()
+	defer s.m.Unlock()
+	twin, err := s.Substrate.CreateTwin(identity, relay, pk)
+	return twin, normalizeNotFoundErrors(err)
+}
+
+// CreateRentContract creates a rent contract
+func (s *SubstrateImpl) CreateRentContract(mnemonic string, nodeID uint32, solutionProviderID *uint64) (uint64, error) {
+	identity, err := s.NewIdentityFromSr25519Phrase(mnemonic)
+	if err != nil {
+		return 0, err
+	}
+	s.m.Lock()
+	defer s.m.Unlock()
+
+	contract, err := s.Substrate.CreateRentContract(identity, nodeID, solutionProviderID)
+	return contract, normalizeNotFoundErrors(err)
+}
+
+// TransferTFTsFromSystem transfer balance to users' account
+func (s *SubstrateImpl) TransferTFTsFromSystem(tftBalance uint64, userMnemonic string, systemMnemonic string) error {
+	// Create identity of user from mnemonic
+	userIdentity, err := s.NewIdentityFromSr25519Phrase(userMnemonic)
+	if err != nil {
+		return err
+	}
+
+	// Create identity of system from mnemonic
+	systemIdentity, err := s.NewIdentityFromSr25519Phrase(systemMnemonic)
+	if err != nil {
+		return err
+	}
+
+	s.m.Lock()
+	defer s.m.Unlock()
+	return s.Transfer(systemIdentity, tftBalance, substrate.AccountID(userIdentity.PublicKey()))
+}
+
+// TransferTFTsToSystem transfer balance to system account
+func (s *SubstrateImpl) TransferTFTsToSystem(tftBalance uint64, userMnemonic string, systemMnemonic string) error {
+	// Create identity of user from mnemonic
+	userIdentity, err := s.NewIdentityFromSr25519Phrase(userMnemonic)
+	if err != nil {
+		return err
+	}
+
+	// Create identity of system from mnemonic
+	systemIdentity, err := s.NewIdentityFromSr25519Phrase(systemMnemonic)
+	if err != nil {
+		return err
+	}
+
+	s.m.Lock()
+	defer s.m.Unlock()
+	return s.Transfer(userIdentity, tftBalance, substrate.AccountID(systemIdentity.PublicKey()))
+}
```

Comment:
This is all related to kubecloud. in grid-client there is no "system" or "user". you can just expose Transfer and move this logic to kubecloud